### ISSUE TITLE
BECU returns incorrect OFX version on profile request

### ISSUE THREAD

**Issue by protestContest** [csingley/ofxtools#134]
A couple things here. BECU's OFX settings have changed (and I'm unable to submit an update at OFX Home). These new settings should work for BECU:

```
url = https://onlinebanking.becu.org/ofx/ofxprocessor.asp
version = 103
fid = 3670
unclosedelements = true
pretty = true
```

However, for `prof` requests, BECU still returns version 1.03 OFX, but incorrectly sets the OFX version header to "203". This causes a parsing issue, since `header.py` identifies the response as v1 OFX, but throws an error on that header due to the type restriction `version = Types.OneOf(102, 103, 151, 160)`.

My local workaround is to just add "203" to the type enum, but that doesn't seem like the right long-term solution. What's the right way to add support for non-compliant responses like this? (Also, BECU, get your shit together.)

**Comment by csingley** [comment]
> I'm unable to submit an update at OFX Home

Yeah [Jesse's not feeling it](http://www.ofxhome.com/ofxforum/viewtopic.php?id=49843).  I'll put your updates into `ofxtools/config/fi.cfg`.  Thanks!

> However, for prof requests, BECU still returns version 1.03 OFX, but incorrectly sets the OFX version header to "203". This causes a parsing issue, since header.py identifies the response as v1 OFX, but throws an error on that header due to the type restriction version = Types.OneOf(102, 103, 151, 160).

What fresh hell is this?!

First... have you tried submitting a bug report with BECU?  In my experience, it's a lot easier dealing with CU staff than commercial banks.  If you knock up a proper bug report including reference to the OFX spec, customer service is likely to pass it along to somebody meaningful in IT, who will probably be at least remotely concerned that if he blows you off completely, you'll show up at the member meeting and chew him (or his SVP) out in person in front of the CEO and the board.s

Failing that... well, the question is how strictly we wish the parser to validate data.

Presumably, Quicken is passing this shit, else it wouldn't happen.  I'm not heavily invested in being stricter than the de facto standard OFX parser, but neither am I interested in inserting a step into the `ofxtools` development cycle that involves checking against Quicken.

Brass tacks - how strict does the validation of this header field need to be?  We never really use the version number.  The distinction between OFXv1 and OFXv2 is quite meaningful, but determined by regex matching rather than the value of the version field.  I kind of picked the particular versions acceptable to both `OFXHeaderV1.version` and `OFXHeaderV2.version`  out of a hat, frankly.  I wonder if replacing both of these with just `version = Types.Integer` would cause any trouble.

**Comment by protestContest** [comment]
> First... have you tried submitting a bug report with BECU? In my experience, it's a lot easier dealing with CU staff than commercial banks. If you knock up a proper bug report including reference to the OFX spec, customer service is likely to pass it along to somebody meaningful in IT, who will probably be at least remotely concerned that if he blows you off completely, you'll show up at the member meeting and chew him (or his SVP) out in person in front of the CEO and the board.

This is a good tip, I'll see what I can do. I kind of assumed contacting support would be a dead end, but it can't hurt to try.

>  I wonder if replacing both of these with just version = Types.Integer would cause any trouble.

Seems pretty reasonable to me. I can try it out sometime soon and make a PR for it, once I get my dev env set up.

**Comment by csingley** [comment]
> I kind of assumed contacting support would be a dead end, but it can't hurt to try.

It's much better to get the problem fixed then to work around it.  This never works with banks (I always try), but at CUs you may be able to leverage their informality.

BECU probably hasn't developed a binder full of scripts where page 1 is "What version of Windows/Quicken are you running? (Flowchart arrow) Sorry we don't support that", and dropped it off on some poor bastards at a call center in Manila that they force me to go through... which is the failure mode with commercial banks.

Still you will probably find it hard to communicate with member services.  My advice is to lie to them, tell them you loathe Quicken and are still limping along on a sunset version of Microsoft Money on a WinXP box you keep airgapped for just this very purpose.  Tell them their server misconfiguration is blowing up MS Money, and point them directly to the config in question (i.e. the `VERSION` field in the OFX header) along with some supporting documentation (i.e. point out that their `PROFRS` claims `VERSION: 103`, and then reference section 2.2 of the spec for OFXv1.0.3, available [here](https://www.ofx.net/downloads/OFX1.0.3.zip))

With any luck they'll flip the scary email straight over to IT, who will realize it's much easier just to fix the stupid misconfig rather than get dragged into some horrible conversation about consortium specifications with an OCD wannabe-engineer member fully clad in tinfoil hat.

Strength to your arm!

>  I can try it out sometime soon and make a PR for it, once I get my dev env set up.

Do me a favor... sit on it, carry the patch yourself for now, and leave this ticket open.  If we can't get the bug fixed at the source, and there's more than 1 or 2 of you experiencing the problem, we'll revisit carrying the patch in the library.

Apparently I did not in fact just pull the acceptable OFX version strings out of a hat... they're listed right there in section 2.2 of the spec; I just merged the spec for 1.0.3 (which Quicken uses) and 1.6 (the latest available).  I do like to stick to the spec!

**Comment by csingley** [comment]
Your updated config for BECU has been merged in [dc72c4d](https://github.com/csingley/ofxtools/commit/dc72c4d4d94863e3efa306ae6067ecc7d88802f6)
